### PR TITLE
test: port libcxx tests for string assign methods

### PR DESCRIPTION
**Pull request by kkajrewicz**

### CODE REVIEW

File: tests/common/iterators_support.hpp [hunk 1]
```diff
@@ -113,7 +113,7 @@ class input_it {
 	using pointer = typename std::iterator_traits<It>::pointer;
 	using reference = typename std::iterator_traits<It>::reference;
 
-	input_it() = delete;
+	input_it() : _it(){};
```
Comment (igchor) [inline tests/common/iterators_support.hpp]: Why you changed this? If this iterator should have non-deleted contructor what about output_it?

File: tests/external/libcxx/basic_string/string.modifiers/string_assign/iterator.pass.cpp [hunk 2]
```diff
@@ -37,33 +189,38 @@ test(nvobj::pool<struct root> &pop, const S &s1, It first, It last,
...
+
+	nvobj::transaction::run(pop, [&] {
+		r->s = nvobj::make_persistent<S>(s1);
+		r->aCopy = nvobj::make_persistent<S>(*r->s);
+	});
+
+	try {
+		r->s->assign(first, last);
+		UT_ASSERT(false);
+	} catch (...) {
```
Comment (igchor) [inline tests/external/libcxx/basic_string/string.modifiers/string_assign/iterator.pass.cpp]: You should probably catch some specific exception.

File: tests/common/iterators_support.hpp (hunk 1 above)
Comment (kkajrewicz) [reply tests/common/iterators_support.hpp]: Default construction of input_it is required by tests included in string_assing/iterator.pass.cpp (IIter() - lines 382, 386, 389). Libcxx implementation of input iterator defines default ctor, but ours not (with no reason as far as I know). Apart of that I will define def. ctor. for output_it as well.

File: tests/external/libcxx/basic_string/string.modifiers/string_assign/iterator.pass.cpp (hunk 2 above)
Comment (kkajrewicz) [reply tests/external/libcxx/basic_string/string.modifiers/string_assign/iterator.pass.cpp]: done

**Pull request merged**